Code/Conviva/Source/Events/ConvivaReporters/AdEventConvivaReporter.swift

```diff
@@ -14,16 +14,22 @@ public class AdEventConvivaReporter: AdEventProcessor, ConvivaAdPlaybackEventsRe
     public let videoAnalytics: CISVideoAnalytics
     public let adAnalytics: CISAdAnalytics
     let storage: ConvivaConnectorStorage
+    private weak var player: THEOplayer?
         
-    init(video: CISVideoAnalytics, ads: CISAdAnalytics, storage: ConvivaConnectorStorage) {
+    init(video: CISVideoAnalytics, ads: CISAdAnalytics, storage: ConvivaConnectorStorage, player: THEOplayer) {
         videoAnalytics = video
         adAnalytics = ads
         self.storage = storage
+        self.player = player
+    }
+    
+    private func calculatedAdType() -> AdTechnology {
+        return (player?.source?.ads != nil) ? .CLIENT_SIDE : .SERVER_SIDE
```

Comment:
we used to have an `integration: AdIntegrationKind` field on the ads. Is it not there anymore?  (that makes a difference between all kind of ad support we have)

Reply:
AdIntegrationKind is 'theo', freewheel', 'google_ima' or 'google_dai'. these are all CSAI integrations I assume. Not sure what their purpose here would be. Could you clarify?
We need to make the difference here between a client side vs server side ad setup.

Reply:
Can't you use:
if the AdIntegrationKind is google_ima --> client-side
if the AdIntegrationKind is google_dai ---> server-side?
Then we don't have to guess the type?

Reply:
No, in our case we are doing a custom ad reporting. So this i not google-ima, nor google-dai. It is SSAI done by the NFL backend (Anvato Stitcher)

As long as we do not support the combo CSAI + SSAI I think the current setup is ok, but once the SSAI stream is combined with a CSAI ad it will have an adsDescription. On the other hand what would conviva expect in that case... as both are correct.